ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts

```diff
@@ -82,23 +81,77 @@ export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IApp
 
-    public sendPageViewPerformanceInternal(pageViewPerformance: PageViewPerformance) {
-        // TODO: Commenting out for now as we this package only supports pageViewTelemetry. Added task 
-        // https://mseng.visualstudio.com/AppInsights/_workitems/edit/1310811
-        /*
-        var pageViewPerformanceData = new Data<PageViewPerformance>(
-            PageViewPerformance.dataType, pageViewPerformance);
-        var pageViewPerformanceEnvelope = new Envelope(pageViewPerformanceData, PageViewPerformance.envelopeType);
-        this.context.track(pageViewPerformanceEnvelope);
-        */
+    public sendPageViewPerformanceInternal(pageViewPerformance: PageViewPerformance, properties?: { [key: string]: any }) {
+        let telemetryItem = TelemetryItemCreator.createItem(pageViewPerformance,
+            PageViewPerformance.dataType,
+            PageViewPerformance.envelopeType,
+            properties);
+
+        this.context.track(telemetryItem);
+    }
+
+    /**
+     * Starts timing how long the user views a page or other item. Call this when the page opens. 
+     * This method doesn't send any telemetry. Call {@link stopTrackTelemetry} to log the page when it closes.
+     * @param name A string that idenfities this item, unique within this HTML document. Defaults to the document title.
+     */
+    public startTrackPage(name?: string) {
+        try {
+            if (typeof name !== "string") {
+                name = window.document && window.document.title || "";
+            }
+
+            this._pageTracking.start(name);
+        } catch (e) {
+            _InternalLogging.throwInternal(
+                LoggingSeverity.CRITICAL,
+                _InternalMessageId.StartTrackFailed,
+                "startTrackPage failed, page view may not be collected: " + Util.getExceptionName(e),
+                { exception: Util.dump(e) });
+        }
+    }
+
+    /**
+     * Logs how long a page or other item was visible, after {@link startTrackPage}. Call this when the page closes. 
+     * @param name The string you used as the name in startTrackPage. Defaults to the document title.
+     * @param url A relative or absolute URL that identifies the page or other item. Defaults to the window location.
+     * @param properties Additional data used to filter pages and metrics in the portal. Defaults to empty. 
+     *                   Any property of type double will be considered a measurement, and will be treated by Application Insights as a metric
+     */
+    public stopTrackPage(name?: string, url?: string, properties?: Object) {
+        try {
+            if (typeof name !== "string") {
+                name = window.document && window.document.title || "";
+            }
+
+            if (typeof url !== "string") {
+                url = window.location && window.location.href || "";
+            }
+
+            this._pageTracking.stop(name, url, properties);
+
+            if (this.config.autoTrackPageVisitTime) {
```

Comment:
is this being supported in the config read currently?

Reply:
This config is coming from the extension's config that is passed as part of the IConfiguration object. Is this what you mean? :)